### ISSUE TITLE
Hosts File Cleanup

### ISSUE THREAD

**Issue by avivonga** [txn2/kubefwd#229]
Hosts file is not restored when a signal is sent to the process during the actual forwarding (still writing to hosts file).
If the signal is sent after the process is done writing to the hosts file, restore is indeed happening.
I think it should be able to clean up no matter when the process is interrupted. Atm, i'm cleaning hosts file myself.
Other than that, great stuff!

**Comment by cjimti** [comment]
@aviv-ga, I agree that some work is to be done here. `kubefwd` tries to err on the side of not removing entries unless it is certain they should be removed. However, please note that you should not "have" to clean up the hosts' file if entires are left in it; they will not harm anything since they point to internal IPs. If you run `kubefwd` again, it will reassign hostnames if there are any duplicates. Personally, if my `kubefwd` process is interrupted for some reason, like my computer going to sleep, then I just restart the same command, let it finish, and control-c it; this ensures it removed what it added.

**Comment by cjimti** [comment]
Fixed in v1.23.0. This release includes graceful shutdown handling that properly cleans up hosts file entries regardless of when the shutdown signal is received.

Additionally, if you have stale entries from previous sessions, use the new `--purge-stale-ips` flag:
```
sudo kubefwd svc -n default --purge-stale-ips
```

See release notes: https://github.com/txn2/kubefwd/releases/tag/v1.23.0
